src/main/java/org/embl/mobie/io/ome/zarr/loaders/N5OMEZarrImageLoader.java

```diff
@@ -324,12 +328,12 @@ private void open() {
         AffineTransform3D transform = new AffineTransform3D();
         if (multiscales.datasets[setupId].coordinateTransformations != null) {
             double[] scale = multiscales.datasets[setupId].coordinateTransformations[0].scale;
-            if (scale != null) {
-                int l = scale.length;
-                if (l > 2) {
-                    transform.scale(scale[l - 3], scale[l - 2], scale[l - 1]);
+            if (scale != null && zarrAxesList != null) {
+                int scalesFirstIndexBackward = scale.length - 1;
+                if (zarrAxes.containsXYZCoordinats()) {
+                    transform.scale(scale[scalesFirstIndexBackward - 2], scale[scalesFirstIndexBackward-1], scale[scalesFirstIndexBackward]);
                 } else {
-                    transform.scale(scale[l - 2], scale[l - 1], 1);
+                    transform.scale(scale[scalesFirstIndexBackward - 1], scale[scalesFirstIndexBackward], 1.0);
```

Comment:
Yes, this looks like my python pseudo code translated to Java ;).

Reply:
Well, it is not that elegant (I think I'll change it a bit once we move metadata logic), but a least it is more understandable now.